rotkehlchen/db/updates.py

```diff
@@ -59,7 +55,56 @@ def __init__(self, msg_aggregator: 'MessagesAggregator', user_db: 'DBHandler') -
             UpdateType.RPC_NODES: self.update_rpc_nodes,
             UpdateType.CONTRACTS: self.update_contracts,
             UpdateType.GLOBAL_ADDRESSBOOK: self.update_global_addressbook,
-        }
+        }  # If we ever change this also change tests/unit/test_data_updates::reset_update_type_mappings  # noqa: E501
+        self.version = pversion.parse(get_current_version().our_version)
+
+    def update_single(
```

Comment:
Made this function contain all the common functionality between all the data update functions